crates/task-impls/src/quorum_vote/drb_computations.rs

```diff
@@ -0,0 +1,105 @@
+use std::collections::BTreeMap;
+
+use hotshot_types::{
+    drb::{compute_drb_result, DrbResult, DrbSeedInput},
+    traits::node_implementation::NodeType,
+};
+use tokio::{spawn, task::JoinHandle};
+
+/// Helper struct to track state of DRB computations
+pub struct DrbComputations<TYPES: NodeType> {
+    /// Stored results from computations
+    results: BTreeMap<TYPES::Epoch, DrbResult>,
+
+    /// Currently live computation
+    task: Option<(TYPES::Epoch, JoinHandle<DrbResult>)>,
+
+    /// Stored inputs to computations
+    seeds: BTreeMap<TYPES::Epoch, DrbSeedInput>,
+}
+
+impl<TYPES: NodeType> DrbComputations<TYPES> {
+    #[must_use]
+    /// Create a new DrbComputations
+    pub fn new() -> Self {
+        Self {
+            results: BTreeMap::new(),
+            task: None,
+            seeds: BTreeMap::new(),
+        }
+    }
+
+    /// If a task is currently live AND has finished, join it and save the result.
+    /// Does not block if a live task has not finished.
+    pub async fn try_join_task(&mut self) {
```

Comment:
Is this function used anywhere? I don't think we'll need this, since whenever we overwrite `self.task` we won't use the result of the existing task anymore.